helpers/index.js

```diff
@@ -340,12 +341,143 @@ async function displayOneOrMultiplePages(interaction, colData, setCustomFields)
+        const completionsFields = resJson.results.map(completionToFields);
+        const fieldWidths = displayFields.map(field => Math.max(
+            ...completionsFields.map(completionFields => {
+                return field === 'Link' ? 4 : completionFields[field].length;
+            }),
+            field.length
+        ));
+        if (mobile) {
+            content += '\n`';
+            content += displayFields.map((field, idx) => field.padEnd(fieldWidths[idx])).join(' ');
+            content += ' `\n';
+            content += completionsFields.map(completionFields => {
+                return '`' + displayFields.map((field, idx) => {
+                    let val = completionFields[field];
+                    let res = val.padEnd(fieldWidths[idx] + val.length - (field === 'Link' ? 4 : val.length));
+                    return field === 'Link' ? '`' + res + '`' : res;
+                }).join(' ') + ' `';
+            }).join('\n');
+        } else {
+            challengeEmbed.addFields(displayFields.map(field => {
+                return {
+                    name: field,
+                    value: completionsFields.map(completionFields => completionFields[field]).join('\n'),
+                    inline: true
+                };
+            }));
+        }
+
+        challengeEmbed.setDescription(content);
+        setOtherDisplayFields(challengeEmbed);
+        return [challengeEmbed, resJson.more];
+    };
+
+    const displayPages = async (initial) => {
+        let [embed, more] = await createPage();
+
+        await interaction.editReply({embeds: [embed], components: !initial || more ? [multipageButtons] : [singlepageButtons]});
+
+        const filter = (selection) => {
+            // Ensure user clicking button is same as the user that started the interaction
+            if (selection.user.id !== interaction.user.id) {
+                return false;
+            }
+            // Ensure that the button press corresponds with this interaction and wasn't
+            // a button press on the previous interaction
+            if (selection.message.interaction.id !== interaction.id) {
+                return false;
+            }
+            return true;
+        };
+
+        const collector = interaction.channel.createMessageComponentCollector({
+            filter,
+            componentType: ComponentType.Button,
+            time: 60000
+        });
+
+        collector.on('collect', async (buttonInteraction) => {
+            collector.stop();
+            buttonInteraction.deferUpdate();
+
+            let offset = parseInt(searchParams.get('offset') ?? '0');
+
+            switch (buttonInteraction.customId) {
+                case 'mobile':
+                    mobile = !mobile;
+                    await displayPages(initial);
+                    break;
+                case '-1':
+                    searchParams.set('offset', Math.max(offset - NUM_ROWS, 0));
```

Comment:
I like the functionality of wrapping around back to the end. Same with below, it should wrap around to the start.